cargo/src/bin/cargo_audit.rs

```diff
@@ -24,6 +24,75 @@
 #![deny(clippy::manual_let_else)]
 #![allow(clippy::unreachable)]
 
-fn main() {
-    println!("Todo-uwu");
+use obs_service_cargo::audit;
+use obs_service_cargo::audit::Src;
+use std::error::Error;
+use std::fmt;
+use std::fmt::Debug;
+use std::fmt::Display;
+
+struct AuditError {
+    msg: String,
+}
+
+impl Debug for AuditError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.msg)
+    }
+}
+
+impl Display for AuditError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.msg)
+    }
+}
+
+impl Error for AuditError {}
+
+fn main() -> Result<(), AuditError> {
+    if let Some(arg) = std::env::args().nth(1) {
```

Comment:
Interesting approach? 🤔

Reply:
You mean the std::env? nah it's just boilerplate. Just testing some stuff. It will be removed later on a future commit. Pseudocode may look like:

```rust
let args = audit::Opts::parse()?;

if !args.srctar.is_none() && !args.srcdir.is_none() {
if let Some(uwutar) = args.srctar {
   // Code logic uwu...
   Ok(())
}
info!("SrcTar option is not used! Checking SrcDir!");
if let Some(uwudir) = args.srcdir {
   // Code logic uwu...
   Ok(())
}
} else {
  // read _service file and do some magic
  Ok(())
}

Reply:
Yes. No indents 👀 hard to type here in github

Reply:
All good, just wondered why you didn't go with clap was all :)